src/Products/MailHost/MailHost.py

```diff
@@ -352,24 +353,45 @@ class MailHost(Persistent, MailBase):
 
 
 def _encode(body, encode=None):
-    """Manually sets an encoding and encodes the message if not
-    already encoded."""
-    if encode is None:
-        return body
+    """
+    Manually sets an encoding and encodes the message if not already encoded.
+
+    If the message already has an encoding header set that does not match the
+    given encoding, an error is raised.
+
+    If there is no encoding header yet but an encoding is given manually, the
+    message is encoded accordingly.
+
+    If the (included or manually set) encoding is 8bit, the message is returned
+    as bytes because otherwise smtplib.sendmail will try to encode it expecting
+    only seven bits.
+    """
     mo = message_from_string(body)
     current_coding = mo['Content-Transfer-Encoding']
-    if current_coding == encode:
-        # already encoded correctly, may have been automated
-        return body
-    if mo['Content-Transfer-Encoding'] not in ['7bit', None]:
+
+    if encode is None:
+        encode = current_coding
+
+    if current_coding and encode != current_coding:
         raise MailHostError('Message already encoded')
-    if encode in ENCODERS:
-        ENCODERS[encode](mo)
-        if not mo['Content-Transfer-Encoding']:
-            mo['Content-Transfer-Encoding'] = encode
-        if not mo['Mime-Version']:
-            mo['Mime-Version'] = '1.0'
-    return mo.as_string()
+
+    if encode:
+        if encode not in ENCODERS:
+            raise MailHostError('Unknown encoding')
+
+        if not current_coding:
+            ENCODERS[encode](mo)
+            if not mo['Content-Transfer-Encoding']:
+                mo['Content-Transfer-Encoding'] = encode
+            if not mo['Mime-Version']:
+                mo['Mime-Version'] = '1.0'
+
+    if encode == '8bit':
+        import pdb
+        pdb.set_trace()
```

Comment:
This looks not like intentional.

Reply:
Sorry. As the title says, this is still WIP. In particular, the tests fail and passing the given `Policy` to `as_bytes()` does not actually work with non-ASCII content, which defeats the purpose - I'll have to read more documentation. I just wanted to push what I got so far, in case someone sees that I am headed in completely the wrong direction and can correct me.

I now see that there is a possibility to convert a PR to draft. I have not used this yet, but I guess this is appropriate here.

Reply:
All good! :-)

I am curious how this all turns out. We run on Zope 4 for quite a while, and we are heavy users of MailHost and we did not experience any problems so far.